src/components/search/FacetList.jsx

```diff
@@ -0,0 +1,112 @@
+import React, {
+  useCallback, useContext, useEffect, useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import { Collapsible, Input } from '@edx/paragon';
+import { connectRefinementList } from 'react-instantsearch-dom';
+import classNames from 'classnames';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faChevronUp, faChevronDown } from '@fortawesome/free-solid-svg-icons';
+import OutsideClickHandler from 'react-outside-click-handler';
+import { AppContext } from '@edx/frontend-platform/react';
+
+import './styles/FacetList.scss';
+
+const BaseFacetList = ({
+  title,
+  attribute,
+  items,
+  currentRefinement,
+  refine,
+}) => {
+  const { enterpriseConfig: { branding } } = useContext(AppContext);
+  const [isOpen, setIsOpen] = useState(false);
+
+  useEffect(() => {
+    function checkKeyAndCloseIfEsc({ key }) {
+      if (key === 'Escape') {
+        setIsOpen(false);
+      }
+    }
+    document.addEventListener('keydown', checkKeyAndCloseIfEsc);
+    return () => { // on unmount, remove event listener
+      document.removeEventListener('keydown', checkKeyAndCloseIfEsc);
+    };
+  }, []);
+
+  const handleOutsideClick = () => {
+    if (isOpen) {
+      setIsOpen(false);
+    }
+  };
+
+  const renderItems = useCallback(
+    () => {
+      if (!items.length) {
+        return (
+          <p>No options found.</p>
+        );
+      }
+      return (
+        <ul className="list-group">
+          {items.map((item, index) => (
+            <li key={item.label} className="list-group-item border-0">
+              {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+              <label htmlFor={`${attribute}-${index}`}>
+                <Input
+                  type="checkbox"
+                  id={`${attribute}-${index}`}
+                  checked={item.isRefined}
+                  onChange={() => refine(item.value)}
+                />
+                <span className={classNames('facet-item-label', 'ml-1', { 'is-refined': item.isRefined })}>
+                  {item.label}
+                </span>
+                <span
+                  className="badge badge-pill text-white ml-2"
+                  style={{ backgroundColor: branding.search.backgroundColor }}
```

Comment:
replace this inline style with `bg-brand-primary` once https://github.com/edx/frontend-app-learner-portal-enterprise/pull/36 is merged.